xml_struct_tests/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use quick_xml::Writer;
+use thiserror::Error;
+use xml_struct::XmlSerialize;
+
+pub fn serialize_value_as_element<T>(value: T, root_name: &str) -> Result<String, TestError>
+where
+    T: XmlSerialize,
+{
+    let buf = Vec::default();
+    let mut writer = Writer::new(buf);
+
+    value.serialize_as_element(&mut writer, root_name)?;
+
+    let out = String::from_utf8(writer.into_inner())?;
+
+    Ok(out)
+}
+
+pub fn serialize_value_children<T>(value: T) -> Result<String, TestError>
+where
+    T: XmlSerialize,
+{
+    let buf = Vec::default();
+    let mut writer = Writer::new(buf);
+
+    value.serialize_child_nodes(&mut writer)?;
+
+    let out = String::from_utf8(writer.into_inner())?;
+
+    Ok(out)
+}
+
+#[derive(Debug, Error)]
+pub enum TestError {
```

Comment:
AFAIK using `#[from]` already generates the relevant From conversions and implicitly adds the `source`, so I'm unsure this is actually needed..?